### PR TITLE
Stop sharing plugin versions between the WAR and tests

### PR DESCRIPTION
**Pull request by basil**
As suggested in [#5487 (comment)](https://github.com/jenkinsci/jenkins/pull/5487#discussion_r631902800), removing some POM properties to decouple the versions we bundle from the versions we use for functional tests. I am retaining the current version of Matrix Authorization Strategy for the WAR but upgrading it to the latest version for the tests. As a bonus, I also added the test scope to all test dependencies.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
Sounds about right

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
